Compare time instead of dates

punkeel opened the pull request:
The former code would fail (imo) at 11PMxx: the asleep variable
contains yesterday's 11PM (which is after now, √) but awake would be
today's 9AM, which is not before now -> the && fails

PS/ Contribution to not-useful projects is my favorite hobby 💯

punkeel committed:
Compare time instead of dates

The former code would fail (imo) at 11PMxx: the asleep variable
contains yesterday's 11PM (which is after now, √) but awake would be
today's 9AM, which is not before now -> the && fails

notwaldorf commented on app/do.js:
```diff
@@ -5,18 +5,20 @@ ipc.on('show', function (event, message) {
+
 function doItGetTheTimeDoItNow() {
-  var now = moment().tz("Asia/Taipei");
+  var now = minutesOfDay( )moment().tz("Asia/Taipei"));
```
it looks like there's an extra bracket here?

punkeel replied:
You're right

notwaldorf commented:
@PunKeel Oh thank god, timezone math is my least favourite thing! 😵

Also I think I pushed some things and you need a rebase. Sorry!

punkeel committed:
Fix mistakes

One formatting mistake & one auto-completion mistake. This makes two mistakes.

punkeel committed:
Merge master into patch-1

punkeel commented:
@notwaldorf Rebased! 👍 

If I remember correctly, GitHub lets you merge my commits into one, so I won't do it

notwaldorf commented on app/do.js:
```diff
@@ -5,18 +5,20 @@ ipc.on('show', function (event, message) {
+
 function doItGetTheTimeDoItNow() {
-  var now = moment().tz('Asia/Taipei');
+  var now = minutesOfDay(moment().tz('Asia/Taipei'));
```
Oh so this `now` is used later to display the time (with a `now.format()`). I think you need a different variable either here or later.

punkeel committed:
Fix display

Fix variable now (used in display), add a missing ; and use innerText instead of innerHTML.
Who knows? Emojis are evil and might trigger XSS.

notwaldorf commented:
I'll just merge it and fix the string in a follow up PR. Thanks!!

punkeel commented:
You're welcome! 
Sorry about the multiple fails.

PS/ Isn't the string fixed with this line: https://github.com/notwaldorf/is-mu-an-asleep/pull/1/commits/599e262b9067a1a7dfaa4f058b8b2eacd013cc91#diff-93b3910207774b01d21c9e5b050793f9R14 ? 🤔

notwaldorf commented:
@PunKeel Yeah! I noticed that after merging it. But then the SVG `innerText` doesn't work (you need `innerHTML`), but it was a quick fix. Yay!

punkeel commented:
@notwaldorf Worked in chrome, I assumed it would work on Electron too. My bad.
(I'm still trying to download Electron, slow connection doesn't help)

Yay! \o/
